src/nimlsp.nim

```diff
@@ -118,8 +118,24 @@ proc getProjectFile(file: string): string =
       discard
     path = dir
 
+proc findProjectNimFile*( pkg: string): string =
```

Comment:
How does this differ from the existing `getProjectFile`? Why is this better?

Reply:
it meat to find nim file through project root then pass to nimsuggest.

Reply:
not sure better or not , it assuming the project main file named same as project root dir name. just few lines